toolkit/tools/pkg/imagecustomizerlib/telemetry.go

```diff
@@ -0,0 +1,78 @@
+package imagecustomizerlib
+
+import (
+	"context"
+	"net"
+	"runtime"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/logger"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace"
+	"go.opentelemetry.io/otel/trace/noop"
+)
+
+var (
+	Tracer trace.Tracer
+
+	TracerName = "imagecustomizer"
+
+	TelemetryServerAddress = "localhost:4317"
```

Comment:
While we generally recommend that people run inside a container, there are some who don't. Also, dev testing is usually done using `go run` and `go test`. So, picking a fixed port number (`4317`) is probably not a good idea in case it is already being used for something else. This should probably be a CLI parameter.

Reply:
Combining with your other comment about the `OTEL_TRACES_EXPORTER` env var, maybe it'll be better if that env var is meant to hold the server address, and instead of adding a CLI parameter we just use the env var value? What do you think about that?

Reply:
Yes, I added this comment before I discovered the env variables as an option.